storage/sql/coresql/db/storage.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package db
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian/storage/sql/coresql/wrapper"
+	"github.com/google/trillian/storage/sql/mysql"
+)
+
+// OpenDB opens, and creates a wrapper for, the SQL databases that we know about. The wrapper
+// encapsulates things that differ between database implementations and abstracts away
+// from the SQL queries.
+func OpenDB(driver, dbURL string) (wrapper.DBWrapper, error) {
```

Comment:
I'll have another look at it but I previously couldn't avoid import cycles unless it was broken out.